### ISSUE TITLE
Feature Request:  Sensor for pool cover

### ISSUE THREAD

**Issue by larryboy44** [dwradcliffe/intellicenter#9]
My Pentair controller has an add-on board for the powered pool cover so if the cover is closed, and freeze protection is activated, it won't switch on the laminars.

Is it possible to have a sensor that returns the state of the cover?  I'd like to use it for an automation to turn on the pool lights if open.

**Comment by snakeeyes5** [comment]
Just came here looking for the same thing. I have a few automations I was hoping to leverage using the cover status.

**Comment by dwradcliffe** [comment]
I’m happy to add this but I’ll need some debug info since I don’t have this
on my system.

---
David Radcliffe

**Comment by snakeeyes5** [comment]
Cool - can you give some instructions on what you need?

**Comment by dwradcliffe** [comment]
Let's start with the diagnostic info you can get from the integration. Just go to Settings > Devices > Pentair Intellicenter > Download Diagnostics.

<img width="335" alt="Monosnap Settings – Home Assistant 2024-08-31 17-37-02" src="https://github.com/user-attachments/assets/a4728806-50a3-4e8b-8b24-7de3dcc79345">

**Comment by larryboy44** [comment]
I'm also interested in this functionality.

Did you get the diagnostics from SnakeEyes or shall I send it?

Thanks.

**Comment by dwradcliffe** [comment]
I have not gotten any diagnostics from anyone yet.

**Comment by larryboy44** [comment]
Ok, I can get them to you.

But, I don't have a Download Diagnostics button as you show in your image.  I have Enable Debug Logging(and then Disable).  According the system I do have the latest version (1.4.0a3) but it's older than you show in your image above.

**Comment by dwradcliffe** [comment]
To get to that view, either click on the device link, or you can directly download diagnostics from the three-dot menu.

<img width="1170" alt="Monosnap Settings – Home Assistant 2024-09-18 21-19-32" src="https://github.com/user-attachments/assets/49a14856-c105-4526-a34e-06deee8831c8">

**Comment by larryboy44** [comment]
and what I get...  no download option.  What am I missing  :-(

[image: McKee.Pentair.png]

**Comment by larryboy44** [comment]
![McKee Pentair](https://github.com/user-attachments/assets/22ae6032-8df1-42cf-bb22-f01140d9064e)

**Comment by dwradcliffe** [comment]
Can you confirm what version of the integration is downloaded (from HACS page):
<img width="552" alt="Monosnap HACS – Home Assistant 2024-09-21 13-05-06" src="https://github.com/user-attachments/assets/412b1f4e-a4b8-45a9-814d-6fb57c475f26">

Also what version of HA?

**Comment by larryboy44** [comment]
Pentair Intellicenter for Home Assistant  v1.5

Home Assistant Core 2024.   v7.2
Home Assistant Operating System  v12.4

**Comment by dwradcliffe** [comment]
Looks like you might be using the other fork (`jlvaillant`) not my fork?
To switch:
Don't remove the device from HA. First remove the integration from HACS. Do not restart. Add my repo to HACS. Then when you restart it should just load up with the new code.

**Comment by larryboy44** [comment]
Got it.

Log file attached.

**Comment by larryboy44** [comment]
[config_entry-intellicenter-193efda1ade778c683dac9e69ff69327.json](https://github.com/user-attachments/files/17085306/config_entry-intellicenter-193efda1ade778c683dac9e69ff69327.json)

**Comment by larryboy44** [comment]
I see nothing in the log file that specifically mentions a cover, but there are some oddly named items.  The cover was closed when I retrieved this log.  I'll open it and retrieve the log again and maybe an entry will be different.

**Comment by larryboy44** [comment]
A second log file but this time the cover is open. 
[config_entry-intellicenter-193efda1ade778c683dac9e69ff69327 (1).json](https://github.com/user-attachments/files/17085369/config_entry-intellicenter-193efda1ade778c683dac9e69ff69327.1.json)

**Comment by larryboy44** [comment]
Have you had an opportunity to look into this?

**Comment by dwradcliffe** [comment]
Those don't seem to show anything we can use. :(
@jonoberheide can you upload debug info to see if maybe there's something in yours?

**Comment by jonoberheide** [comment]
@dwradcliffe Two things:

(1) I realized I was using the wrong github repo. My HACS said "dwradcliffe", but it was jlvaillant's repo. Probably because you're listed in the manifest.json. I noticed because there was no "Download Diagnostics" on my device page.

(2) Pentair just released a firmware update last night. One of the changelog lines is "Covers feature". Perhaps that will be relevant here. I'm really surprised the cover status doesn't seem to be shown anywhere in the controller/app, so hopefully they're changing that.

https://www.pentair.com/en-us/pool-spa/education-support/homeowner-support/software-downloads/intellicenter-control-system.html

Attached is the diagnostics.

[config_entry-intellicenter-01JV3W24BKMK2R3WZGGSHYVTZH.json](https://github.com/user-attachments/files/20244457/config_entry-intellicenter-01JV3W24BKMK2R3WZGGSHYVTZH.json)

**Comment by dwradcliffe** [comment]
I still don't see anything about the cover in the dump. This was with the latest firmware?

**Comment by jonoberheide** [comment]
Yes, 3.004. It looks like the CVR01 object is of type EXTINSTR and subtype COVER, but I only seem a couple references to that in the pyintellicenter code (eg. EXTINSTR_ATRIBUTES).

**Comment by dwradcliffe** [comment]
Where do you see `CVR01`? It isn't in the file you uploaded.

**Comment by jonoberheide** [comment]
Sorry, I attached it to the issue in the other repo. Re-adding here.

[home-assistant_intellicenter_2025-05-13T03-27-58.622Z.log](https://github.com/user-attachments/files/20320349/home-assistant_intellicenter_2025-05-13T03-27-58.622Z.log)

**Comment by dwradcliffe** [comment]
Ok! Try the latest version and see if it works. It's hard for me to test but I gave it a good guess 😆

**Comment by jonoberheide** [comment]
Hrm, doesn't seem to have added any cover entities after upgrading and reloading...

[home-assistant_intellicenter_2025-05-23T02-27-20.892Z.log](https://github.com/user-attachments/files/20403066/home-assistant_intellicenter_2025-05-23T02-27-20.892Z.log)

**Comment by jonoberheide** [comment]
Interesting to note that the Pentair app now displays the pool cover. I think 3.004 firmware must have exposed the support and then a recent Pentair app update is now consuming it:

![Image](https://github.com/user-attachments/assets/eba793ce-0e91-491e-8ea0-66e8b7b45cf8)
